### PR TITLE
Fix the snackbar notices position

### PR DESCRIPTION
**Pull request by youknowriad**
This PR fixes another small regression caused by #18044 where the snackbar notices were positioned at the top instead of the bottom left of the page.

Basically the `edit-post-layout__content` className has been removed. Fortunately, this is the only place it was used. I'm also removing some useless styles I found.

I'm also adding a "Needs dev note" label to document this change in the next WordPress release.

### CODE REVIEW

File: packages/edit-post/src/components/layout/style.scss [hunk 1]
```diff
@@ -110,30 +110,3 @@
...
-	// then pause, then keep scrolling downwards, the browser doesn't try to scroll
-	// the parent element, usually invoking a "bounce" effect and then preventing you
-	// from scrolling upwards until you pause again.
-	// This is only necessary beyond the small breakpoint because that's when the scroll container changes.
-	@include break-small() {
-		overscroll-behavior-y: none;
-	}
-
-	// Pad the scroll box so content on the bottom can be scrolled up.
-	padding-bottom: 50vh;
```
Comment (ellatrix) [inline packages/edit-post/src/components/layout/style.scss]: Why is this removed?

File: packages/edit-post/src/components/layout/style.scss [hunk 2]
```diff
@@ -110,30 +110,3 @@
 		font-size: $default-font-size;
 	}
 }
-
-.edit-post-layout__scrollable-container {
-	// On mobile the main content (html or body) area has to scroll.
```
Comment (ellatrix) [inline packages/edit-post/src/components/layout/style.scss]: Have you tested this on mobile?

Comment (ellatrix) [reply packages/edit-post/src/components/layout/style.scss]: Cc @jasmussen. I'm not sure who originally added this.

Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: This class is not used anywhere.

Comment (jasmussen) [reply packages/edit-post/src/components/layout/style.scss]: Oh wow, a lot of red here which I remember painstakingly adding. But yeah, if the class is not present anymore, this has either regressed long ago, or is completely unnecessary. I'll take a look!

Comment (ellatrix) [reply packages/edit-post/src/components/layout/style.scss]: When (or in what commit) was the class removed?

File: packages/edit-post/src/components/layout/style.scss (hunk 1 above)
Comment (ellatrix) [reply packages/edit-post/src/components/layout/style.scss]: I believe this part in necessary for the typewriter experience.

File: packages/edit-post/src/components/layout/style.scss (hunk 2 above)
Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: In this refactor #18044

Comment (ellatrix) [reply packages/edit-post/src/components/layout/style.scss]: Ok, so it wouldn't have regressed long ago, the class was removed two days ago.

Comment (jasmussen) [reply packages/edit-post/src/components/layout/style.scss]: It looks like the bulk of this behavior has moved to `editor-regions` and looks like this:

```
.edit-post-editor-regions__body {
	flex-grow: 1;
	display: flex;

	// On Mobile the header is fixed to keep HTML as scrollable.
	@include break-medium() {
		overflow: auto;
	}
}

.edit-post-editor-regions__content {
	flex-grow: 1;

	// On Mobile the header is fixed to keep HTML as scrollable.
	@include break-medium() {
		overflow: auto;
	}
}
```

I'll take a look at how it scrolls on iOS, which I think still needs things like -webkit-overflow-scrolling: touch;, and Firefox which I believe we added overscroll-behavior-y: none; for.

File: packages/edit-post/src/components/layout/style.scss (hunk 1 above)
Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: I guess we should restore it then, but it's not clear how it's related since it's very detached from the typewriter experience. Maybe it should be added in JS or in the typewriter component.

Also it's a separate regression from this PR I think.

Comment (ellatrix) [reply packages/edit-post/src/components/layout/style.scss]: It's not strictly needed for the typewriter experience. It just makes it nicer. It's styling and it can be adjusted. The comment says what it is: extra space on the bottom of the page so it can be scrolled up more. I'm not sure what's a better place for it.

Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: Maybe still this file then by overriding the editor regions style, EditorRegions is meant to be something generic that will be used outside the editor (widgets, ... any page that has the same global layout than the editor) so I don't think it makes sense there.

File: packages/edit-post/src/components/layout/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/edit-post/src/components/layout/style.scss]: Good news, this branch and master seems to work the same in iOS as it did before, with the correct element scrolling:

master

I think it might be worth it to restore some of these comments before the overflow rule quoted above, just because it's so easy to assume Mobile Safari works well when Safari works so well, when in fact Mobile Safari is horrendously bad and light years behind. The "wall of text" helps remind people to maybe test on the xcode simulator.

I'll be touching the top toolbar PR later today, I'll see if I can add it there.

Comment (jasmussen) [reply packages/edit-post/src/components/layout/style.scss]: Looks like overscroll-behavior-y: none isn't necessary either, scrolling works in <600px in chrome and Firefox, as was the reason for the rule in the first place:

https://github.com/WordPress/gutenberg/pull/10539

File: packages/edit-post/src/components/layout/style.scss (hunk 1 above)
Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: @ellatrix would you min adding the fix to this PR or a separate one. I'm personally not sure what impact this have and to be honest, the type writer experience feels good even without it for me.

Comment (gwwar) [reply packages/edit-post/src/components/layout/style.scss]: I don't really see any difference with these rules added/removed, but I also not sure when this class is added. Querying the DOM in the post editor returns 0 nodes for me. Overall reading this, removal of these rules feel unrelated to the fix here. Is this blocking @ellatrix ?

**Review by gwwar — APPROVED** [review]
Thanks @youknowriad I verified that this fixes the snackbar position. Maybe let's drop changes to `edit-post-layout__scrollable-container` since I didn't see how this was related in testing. If it's not actually needed, lets split that out to another janitorial. 

Screen Shot 2019-11-22 at 10 23 45 AM

File: packages/edit-post/src/components/layout/style.scss (hunk 1 above)
Comment (youknowriad) [reply packages/edit-post/src/components/layout/style.scss]: These rules don't do anything at the moment because that class don't exist in the DOM. i think the fix should be done separately as this is just dead code for now. And the fix is not to bring the whole style back but just the padding bottom if I understand properly.
